### PR TITLE
Individual questionnaire linked

### PR DESCRIPTION
**Pull request by AdeHarrison**
# Motivation and Context
Respondent's need to be able to request individual paper questionnaire forms from a field officer

# What has changed
An individual paper questionnaire request creates a new HI case based on the HH case
Refactored IT tests to be more realistic so that UAC/QID's are created from unaddressed uac messages

# How to test?
Build local docker image of this branch and run with docker dev
Run tests
Run acceptance tests from [this branch](https://github.com/ONSdigital/census-rm-acceptance-tests/pull/119)

# Links
[Trello](https://trello.com/c/6mZA6Ynw/1040-rmc-142-part-2-handle-individual-pq-linked-event-8)

# Screenshots (if appropriate):

### CODE REVIEW

**Comment by AdeHarrison** [comment]
[RMC-142 (part 2) Handle Individual PQ Linked Event (8)](https://trello.com/c/6mZA6Ynw/1040-rmc-142-part-2-handle-individual-pq-linked-event-8)

File: src/main/java/uk/gov/ons/census/casesvc/service/FulfilmentRequestService.java [hunk 1]
```diff
@@ -61,7 +61,7 @@ public void processFulfilmentRequest(ResponseManagementEvent fulfilmentRequest)
     }
   }
 
-  private Case prepareIndividualResponseCaseFromParentCase(Case parentCase) {
+  public Case prepareIndividualResponseCaseFromParentCase(Case parentCase) {
```
Comment (LukeLoze1) [inline src/main/java/uk/gov/ons/census/casesvc/service/FulfilmentRequestService.java]: Move this function into CaseService as it's being used in multiple places

File: src/main/java/uk/gov/ons/census/casesvc/utility/QuestionnaireTypeHelper.java [hunk 2]
```diff
@@ -54,4 +54,18 @@ public static int calculateQuestionnaireType(String treatmentCode) {
   public static boolean isQuestionnaireWelsh(String treatmentCode) {
     return (treatmentCode.startsWith("HH_Q") && treatmentCode.endsWith("W"));
   }
+
+  public static boolean isIndividualQuestionnaireType(String questionnaireId) {
```
Comment (LukeLoze1) [inline src/main/java/uk/gov/ons/census/casesvc/utility/QuestionnaireTypeHelper.java]: We've tended to follow a pattern of Maps or Sets for this sort of thing, 
i.e.
  static final Set<String> individualThings =  new HashSet<>(
          Arrays.asList("21", "22", "23", "24"));

  if( individualThings.contains(questionnaireId.substring(0, 2) ) {
        do x;

  See example in fulfilmentRequestService.

File: src/main/java/uk/gov/ons/census/casesvc/service/FulfilmentRequestService.java (hunk 1 above)
Comment (AdeHarrison) [reply src/main/java/uk/gov/ons/census/casesvc/service/FulfilmentRequestService.java]: done but knock-on effect with tests, out of scope and should be in tech backlog

File: src/main/java/uk/gov/ons/census/casesvc/service/QuestionnaireLinkedService.java [hunk 3]
```diff
@@ -17,19 +18,34 @@
 
   private final UacService uacService;
   private final CaseService caseService;
+  private final FulfilmentRequestService fulfilmentRequestService;
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/census/casesvc/service/QuestionnaireLinkedService.java]: This looks unused now

Comment (AdeHarrison) [reply src/main/java/uk/gov/ons/census/casesvc/service/QuestionnaireLinkedService.java]: removed

**Pull request merged**